### PR TITLE
Move documentation to sphinx

### PR DESCRIPTION
**Pull request by martin-schulze-vireso**
This is a first step towards hosting version specific documentation as discussed in #318.

- [x] how to update the documentation version automatically with a new release?
- [x] remove warning about unreleased features

- [x] I have reviewed the [Contributor Guidelines][contributor].
- [x] I have reviewed the [Code of Conduct][coc] and agree to abide by it

[contributor]: https://github.com/bats-core/bats-core/blob/master/docs/CONTRIBUTING.md
[coc]:         https://github.com/bats-core/bats-core/blob/master/docs/CODE_OF_CONDUCT.md

### CODE REVIEW

**Comment by martin-schulze-vireso** [comment]
I just c&p'd most of the readme over into specific subsections of the Sphinx based documentation and replaced the hardcoded `bats --help` output with dynamic output from a run of the local version. We would still need to create the right environment with sphinx + recommonmark + sphinx-contrib on it and host the output of `cd docs; make html;` somewhere on the internet to make it readable.

**Review by sublimino — APPROVED** [review]
LGTM, just re-signed and awaiting build

**Comment by martin-schulze-vireso** [comment]
This is part of getting the documentation onto github pages or read the docs. As is, we would throw away the documentation. Do we have a decision how to display the results @sublimino? We also need to deal with the versioning.

**Comment by sublimino** [comment]
My only preference is simplicity, is GitHub pages easier perhaps?

**Comment by martin-schulze-vireso** [comment]
A first try of this can be seen here: https://martin-schulze-vireso.github.io/bats-core/master/

**Comment by martin-schulze-vireso** [comment]
So we won't get the deploy stage green at this point because sphinx will try to build the refs master and version tags but there are non with sphinx documentation so far. Additionally, according to the [documentation](https://docs.travis-ci.com/user/deployment/pages/#setting-the-github-token) we need the environment variable GITHUB_TOKEN in travis to be filled with a [personal access token](https://help.github.com/articles/creating-an-access-token-for-command-line-use/) with the `public_repo` scope.

**Comment by martin-schulze-vireso** [comment]
I moved this over to read the docs: https://bats-core.readthedocs.io/en/latest/

This leaves us with the following steps:
- [ ] migrate to the actual bats-core/bats-core project (instead of my fork)
- [ ] add other maintainers' accounts on read the docs
- [ ] add an automation for activating new versions

**Comment by martin-schulze-vireso** [comment]
This PRs documentation can be seen at https://bats-core--346.org.readthedocs.build/en/346/

**Comment by sublimino** [comment]
This is ready to merge from my perspective, but there's an unexpected [Bash 5-only test failure](https://travis-ci.org/github/bats-core/bats-core/jobs/749136062), which I hope isn't a breaking Bash change but haven't yet debugged.

**Comment by martin-schulze-vireso** [comment]
From a first glance it looks like GNU parallel is not being found.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 language: bash
 
+stages:
+  - test
```
Comment (martin-schulze-vireso) [inline .travis.yml:4]: This leftover change can be removed.

**Comment by martin-schulze-vireso** [comment]
Some days ago bash:5.1 has been released, which shows this error. bash:5.0 does not. I bisected on 5.1 to this commit 3ed95c415ae164e2c014691fa7417646248100b2. I am looking into the problem but it is unrelated to this PR since it was already present on older commits with bash:5.1

**Comment by martin-schulze-vireso** [comment]
@sublimino I think this is ready now.

**Comment by sublimino** [comment]
Thanks @martin-schulze-vireso, I dug through https://github.com/bminor/bash/blob/master/CHANGES but couldn't see anything jump out. Good find.

And great work on getting functional documentation for the project!! :fireworks: :tada:

**Comment by sublimino** [comment]
(I had to sign your commits to merge to master @martin-schulze-vireso)

**Comment by martin-schulze-vireso** [comment]
Thanks @sublimino, signing is not part of my default workflow, so I sometimes forget the final signing rebase.

**Comment by jasonkarns** [comment]
> Thanks @sublimino, signing is not part of my default workflow, so I sometimes forget the final signing rebase.

@martin-schulze-vireso You can configure git to sign by default in just the bats-core repo, if you don't want to sign by default in _all_ your git repos. `git config commit.gpgsign true`

**Pull request merged**